### PR TITLE
Add dynamic budget and remove debug budget

### PR DESCRIPTION
**Pull request by calniotrey**

### CODE REVIEW

File: common/scripted_triggers/iisc_triggers.txt [hunk 1]
```diff
@@ -67,22 +67,33 @@ iisc_should_build_science_ship = {
...
 
 # this = corp
 iisc_should_destroy_science_ship = {
-	count_owned_ship = {
-		limit = {
-			is_ship_size = science
-		}
-		count > 5		# TODO change
-	}
+	always = no
```
Comment (fireinthedisco) [inline common/scripted_triggers/iisc_triggers.txt:96]: Perhaps TODO, but once the corp is running out of things to survey we may want to reduce the number of science ships.

File: common/scripted_triggers/iisc_triggers.txt [hunk 2]
```diff
@@ -91,22 +102,22 @@ iisc_should_build_builder = {
...
+}
+
+# this = corp
+iisc_has_not_enough_builders = {
 	count_owned_ship = {
 		limit = {
 			is_ship_size = constructor
 		}
-		count < 1		# TODO change
+		count < 5
```
Comment (fireinthedisco) [inline common/scripted_triggers/iisc_triggers.txt:114]: Same as above - 5 constructors feels like a lot for a single corp. Not saying we need to change these now but it should definitely be something we keep an eye on and potentially change later.

File: common/scripted_triggers/iisc_triggers.txt [hunk 3]
```diff
@@ -91,22 +102,22 @@ iisc_should_build_builder = {
...
 
 # this = corp
 iisc_should_destroy_builder = {
-	count_owned_ship = {
-		limit = {
-			is_ship_size = constructor
-		}
-		count > 5		# TODO change
-	}
+	always = no
```
Comment (fireinthedisco) [inline common/scripted_triggers/iisc_triggers.txt:120]: And also like above, once there's not much for the corp to construct we should probably be reducing their number of ships flying around.

**Review by fireinthedisco — APPROVED** [review]
Once again I'm not entirely sure how the budget works yet, but I'm figuring out slowly, but from what I understand this looks good.

It's fundamentally challenging to predict what changes will happen exactly from tweaks to the budget and stuff like this. We will have to see how it works out experimentally and tweak further from there, but for now this is looking like a good foundation.

File: common/scripted_triggers/iisc_triggers.txt [hunk 4]
```diff
@@ -67,22 +67,33 @@ iisc_should_build_science_ship = {
...
+}
+
+# this = corp
+iisc_has_not_enough_science_ships = {
+	OR = {
+		count_owned_ship = {
+			limit = {
+				is_ship_size = science
+			}
+			count < 10
```
[the remark this answers is not in the document]
Comment (calniotrey) [reply common/scripted_triggers/iisc_triggers.txt:80]: Good point. I'll reduce it to 5. Maybe some players can then comment if they think it's too high or too low

File: common/scripted_triggers/iisc_triggers.txt (hunk 1 above)
Comment (calniotrey) [reply common/scripted_triggers/iisc_triggers.txt:96]: Ideally the science ships will have something other to do 🤔

File: common/scripted_triggers/iisc_triggers.txt (hunk 2 above)
Comment (calniotrey) [reply common/scripted_triggers/iisc_triggers.txt:114]: Reduced to 3. Might still be too much. But they should still construct enough stations for the player to notice them

**Pull request merged**